### ISSUE TITLE
Error invoking Enunciate.: java.lang.NullPointerException

### ISSUE THREAD

**Issue by ryangies** [stoicflame/enunciate#986]
Version 2.12.0
Better error messaging appreciated
```
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.12.0:docs (default-cli) on project test-interface: Error invoking Enunciate.: java.lang.NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.12.0:docs (default-cli) on project test-interface: Error invoking Enunciate.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error invoking Enunciate.
    at com.webcohesion.enunciate.mojo.ConfigMojo.execute (ConfigMojo.java:443)
    at com.webcohesion.enunciate.mojo.DocsBaseMojo.execute (DocsBaseMojo.java:99)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
    at com.sun.tools.javac.main.Main.compile (Main.java:553)
    at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
    at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
    at com.webcohesion.enunciate.Enunciate.run (Enunciate.java:667)
    at com.webcohesion.enunciate.mojo.ConfigMojo.execute (ConfigMojo.java:436)
    at com.webcohesion.enunciate.mojo.DocsBaseMojo.execute (DocsBaseMojo.java:99)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at java.util.TreeSet.addAll (TreeSet.java:300)
    at com.webcohesion.enunciate.modules.jackson.model.TypeDefinition.<init> (TypeDefinition.java:129)
    at com.webcohesion.enunciate.modules.jackson.model.ObjectTypeDefinition.<init> (ObjectTypeDefinition.java:38)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.createTypeDefinition (EnunciateJacksonContext.java:292)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared (EnunciateJacksonContext.java:661)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared (EnunciateJacksonContext.java:628)
    at com.webcohesion.enunciate.modules.jackson.javac.ParameterizedJacksonDeclaredType.accept (ParameterizedJacksonDeclaredType.java:51)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions (EnunciateJacksonContext.java:487)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.add (EnunciateJacksonContext.java:412)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared (EnunciateJacksonContext.java:661)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared (EnunciateJacksonContext.java:628)
    at com.webcohesion.enunciate.modules.jackson.javac.ParameterizedJacksonDeclaredType.accept (ParameterizedJacksonDeclaredType.java:51)
    at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions (EnunciateJacksonContext.java:487)
    at com.webcohesion.enunciate.modules.jackson.JacksonModule.addDataTypeDefinitions (JacksonModule.java:206)
    at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.addReferencedDataTypeDefinitions (JaxrsModule.java:259)
    at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.call (JaxrsModule.java:148)
    at com.webcohesion.enunciate.Enunciate.invokeModules (Enunciate.java:890)
    at com.webcohesion.enunciate.EnunciateAnnotationProcessor.process (EnunciateAnnotationProcessor.java:127)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor (JavacProcessingEnvironment.java:794)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs (JavacProcessingEnvironment.java:705)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800 (JavacProcessingEnvironment.java:91)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run (JavacProcessingEnvironment.java:1035)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing (JavacProcessingEnvironment.java:1176)
    at com.sun.tools.javac.main.JavaCompiler.processAnnotations (JavaCompiler.java:1170)
    at com.sun.tools.javac.main.JavaCompiler.compile (JavaCompiler.java:856)
    at com.sun.tools.javac.main.Main.compile (Main.java:523)
    at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
    at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
    at com.webcohesion.enunciate.Enunciate.run (Enunciate.java:667)
    at com.webcohesion.enunciate.mojo.ConfigMojo.execute (ConfigMojo.java:436)
    at com.webcohesion.enunciate.mojo.DocsBaseMojo.execute (DocsBaseMojo.java:99)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

**Comment by ryangies** [comment]
Temporary workaround:
```
diff --git a/jackson/src/main/java/com/webcohesion/enunciate/modules/jackson/javac/ParameterizedJacksonTypeElement.java b/jackson/src/main/java/com/webcohesion/enunciate/modules/jackson/javac/ParameterizedJacksonTypeElement.java
index 278195824..282acb0bb 100644
--- a/jackson/src/main/java/com/webcohesion/enunciate/modules/jackson/javac/ParameterizedJacksonTypeElement.java
+++ b/jackson/src/main/java/com/webcohesion/enunciate/modules/jackson/javac/ParameterizedJacksonTypeElement.java
@@ -80,7 +80,7 @@ public class ParameterizedJacksonTypeElement implements TypeElementAdaptor, HasF
 
   @Override
   public Set<Facet> getFacets() {
-    TreeSet<Facet> facets = null;
+    TreeSet<Facet> facets = new TreeSet<>();
     List<? extends TypeMirror> typeArgs = this.root.getTypeArguments();
     for (TypeMirror typeArg : typeArgs) {
       if (typeArg instanceof DeclaredType) {
```

**Comment by stoicflame** [comment]
Fix has been included in Enunciate 2.12.1.
